query/src/main/scala/filodb/query/exec/SetOperatorExec.scala

```diff
@@ -187,59 +188,175 @@ final case class SetOperatorExec(queryContext: QueryContext,
+
+    // Create a Map of join key as key and value is a two tuple where _1 is all LHS Rvs for the join key and _2
+    // are all RHS Rvs for that same join key
+    val lRvs = lhsResult.foldLeft(Map.empty[Map[Utf8Str, Utf8Str], (Seq[RangeVector], Seq[RangeVector])]) {
+      case (acc, (jk, rvs)) => acc + (jk -> ((rvs, Seq.empty[RangeVector])))
+    }
+
+    val joinedRvs = rhsResult.foldLeft(lRvs) {
+      case (acc, (jk, rvs))  =>
+        val newVal = acc.get(jk).map {
+            // RHS will be Nil
+          case (lhs, rhs) => (lhs, rvs)
+        }.getOrElse((Nil, rvs))
+        acc + (jk -> newVal)
+    }
+
+    case class OrSetOpIteratorWrapper(lhs: Seq[RangeVector], rhs: Seq[RangeVector]) {
+      // Has a side effect where LHS are consumed first and a bit is set to indicate the time instant where there
+      // is at least one non Nan value. the rhs values are then iterated over and emitted only where the given timestep
+      // has no value in any of the LHS timeseries
+      assert(outputRvRange.isDefined, "Expected to see a non null outputRvRange")
+      val (start, end, step) = (outputRvRange.get.startMs, outputRvRange.get.endMs, outputRvRange.get.stepMs)
+      // 64 bits in one word (long), 1 bit per step
+      assert(start == end || step > 0, "Expected to see a non 0 step size when start and end are not same")
+      val wordsRequired = if (start == end) 1 else ((((end - start) / step) >> 6) + 1).toInt
+      val bitSet = new mutable.BitSet(wordsRequired)
+
+      def joined(): Seq[RangeVector] = {
+        val lhsRvs: Seq[RangeVector] = lhs.map(x => {
+          var idx = 0
+          val rows = x.rows()
+          val rowsCursor: RangeVectorCursor = new RangeVectorCursor {
+            val cur = new TransientRow()
+
+            override def hasNext: Boolean = rows.hasNext
+
+            override def next(): RowReader = {
+              val reader = rows.next()
+              // TODO: What is the expected behavior for Histograms?
+              val value = reader.getDouble(1)
+              val res = if (value.isNaN) {
+                Double.NaN
+              } else {
+                bitSet += idx
+                value
+              }
+              idx += 1
+              cur.setValues(reader.getLong(0), res)
+              cur
+            }
+
+            override def close(): Unit = {
+              rows.close()
+            }
+          }
+          IteratorBackedRangeVector(x.key, rowsCursor, x.outputRange)
+        })
+
+        val rhsRvs: Seq[RangeVector] = rhs.map(x => {
+          var idx = 0
+          val rows = x.rows()
+          val rowsCursor: RangeVectorCursor = new RangeVectorCursor {
+            val cur = new TransientRow()
+
+            override def hasNext: Boolean = rows.hasNext
+
+            override def next(): RowReader = {
+              val reader = rows.next()
+              // TODO: What is the expected behavior for Histograms?
```

Comment:
It looks like the previous implementation was agnostic to the row type. Is it enough to just add a `case` here for histograms and set the value accordingly?

Reply:
Previous behavior was also incorrect. :). This behavior is consistent with AND. While I don't necessarily say this is a good thing to do. Im not sure how to deal with histograms yet and hence left a TODO

Reply:
We can leave it as long as any `or` query with a histogram operand can fail (since it wouldn't have explicitly failed before). I can't think of any example that isn't a nonsense query, though.

Reply:
I'll assume it is ok. We can look at any issues that may come up and fix accordingly. 
CC @vishramachandran